pkg/systemd/generate/pods_test.go

```diff
@@ -118,7 +118,7 @@ WantedBy=multi-user.target default.target`
 				StopTimeout:      10,
 				PodmanVersion:    "CI",
 				RequiredServices: []string{"container-1", "container-2"},
-				CreateCommand:    []string{"podman", "pod", "create", "--name", "foo"},
+				CreateCommand:    []string{"podman", "pod", "create", "--name", "foo", "bar=arg with space"},
```

Comment:
I can't quickly figure out whether `strconv.Quote()` handles quotes themselves. Could you change this to `"bar=arg \"with space\""` and have the test itself guarantee that quotes are preserved?

Reply:
Quotes are preserved. I added additional unit tests. Cool with these?
https://github.com/containers/podman/pull/7350/files#diff-1b78823cd7c48fb19dc371e2eb287729R32-R55

Reply:
Yes! Nice!

Reply:
Great, thanks!

Can you /lgtm /hold?